dnsimple/templates.go

```diff
@@ -128,6 +128,22 @@ func (s *TemplatesService) DeleteTemplate(accountID string, templateID string) (
 	return templateResponse, nil
 }
 
+// ApplyTemplate applies a template to the given domain.
+//
+// See https://developer.dnsimple.com/v2/templates/domains/#apply
+func (s *TemplatesService) ApplyTemplate(accountID string, domainID string, templateID string) (*TemplateResponse, error) {
```

Comment:
> I wonder... should we flip the Template and Domain argument?
> 
> Since this is in the template file, and we say Templates.ApplyTemplate I wonder if we should consider "apply this template to this domain".

I prefer not. Let the arguments to be ordered according their appearance in the API path. This communicates the nesting structure of the resources.